onnxruntime/core/optimizer/layer_norm_fusion.cc

```diff
@@ -212,36 +234,82 @@ Status LayerNormFusion::ApplyImpl(Graph& graph, bool& modified, int graph_level,
     }
     nodes_to_remove.push_back(last_add_node);
 
+    // verify axes is contiguous if more than one elements
+    const onnxruntime::NodeAttributes& attributes = reduce_mean_node.GetAttributes();
+    std::vector<int64_t> axes_values, normalized_axes;
+    if (attributes.find("axes") != attributes.end()) {
+      axes_values = RetrieveValues<int64_t>(attributes.at("axes"));
+      if (axes_values.size() > 1) {
+        const int input_dim_size = reduce_mean_node.MutableInputDefs()[0]->Shape()->dim_size();
```

Comment:
Do we need to check if shape() is nullptr? Shape info is not always available. If no inferred shape, we can just use 0 for input_dim_size.

Reply:
on second thought, I'd like to keep it simple. checking axes value is adding unnecessary complexity. So I am dropping the original logic.

---
In reply to: [445825277](https://github.com/microsoft/onnxruntime/pull/4320#discussion_r445825277) [](ancestors = 445825277)